plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java

```diff
@@ -528,16 +527,19 @@ public void publishDiagnostics316(PublishDiagnostics316Param param) {
 			SnykLogger.logError(new InvalidPathException(uri, "couldn't resolve uri " + uri + " to file"));
 			return;
 		}
-		
+
 		populateIssueCache(param, filePath);
 	}
 
-	private void populateFileAndIssueNodes(ProductTreeNode productTreeNode, String folderPath, SnykIssueCache issueCache) {
-		var cacheHashMap = Collections.unmodifiableMap(issueCache.getCacheByDisplayProduct(productTreeNode.getProduct()));
+	private void populateFileAndIssueNodes(ProductTreeNode productTreeNode, String folderPath,
+			SnykIssueCache issueCache) {
+		var cacheHashMap = Collections
+				.unmodifiableMap(issueCache.getCacheByDisplayProduct(productTreeNode.getProduct()));
 		for (var kv : cacheHashMap.entrySet()) {
 			var fileName = kv.getKey();
 			var issues = new ArrayList<>(kv.getValue());
-			if(issues.isEmpty())
+			issues = filterIgnoredIssues(issues);
```

Comment:
Why do we filter here? Shouldn't the filter just filter on the cache when accessing, not when inserting?